libs/common-services/src/main/java/uk/gov/di/ipv/core/library/gpg45/Gpg45Scores.java

```diff
@@ -91,6 +91,41 @@ public List<Evidence> getEvidences() {
         return evidences;
     }
 
+    public List<List<Gpg45Scores.Evidence>> calculateEvidencesRequiredToMeetAProfile(
+            List<Gpg45Profile> profiles) {
+        return profiles.stream()
+                .filter(
+                        profile ->
+                                (activity >= profile.scores.getActivity())
+                                        && (fraud >= profile.scores.getFraud())
+                                        && (verification >= profile.scores.getVerification()))
+                .map(
+                        profile -> {
+                            List<Gpg45Scores.Evidence> acquiredEvidences =
+                                    new ArrayList<>(evidences);
+                            List<Gpg45Scores.Evidence> requiredEvidences =
+                                    new ArrayList<>(profile.scores.getEvidences());
+                            List<Gpg45Scores.Evidence> missingEvidence = new ArrayList<>();
+
+                            while (!requiredEvidences.isEmpty()) {
+                                if (!acquiredEvidences.isEmpty()) {
+                                    if (acquiredEvidences
+                                            .get(0)
+                                            .satisfies(requiredEvidences.get(0))) {
+                                        requiredEvidences.remove(0);
+                                        acquiredEvidences.remove(0);
+                                        continue;
+                                    }
+                                }
+
+                                missingEvidence.add(requiredEvidences.remove(0));
+                            }
+
+                            return missingEvidence;
+                        })
+                .toList();
```

Comment:
Unless you want to argue for the contrary, I still think it's worth making this return the full profile, so this method is a bit more well-defined and reusable.

I think that means:
- Moving the filter from here into the F2F bit - making it clear that you need to have already met the activity/fraud/verification scores to get a profile
- Returning a GPG45Scores here, with activity/fraud/verification scores either zero (if met) or the required level (if unmet)

Hoping that won't take long!

Reply:
Oh I understand. That makes sense, it would be more reusable